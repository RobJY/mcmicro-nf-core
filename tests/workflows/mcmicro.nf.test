nextflow_workflow {

    name "test mcmicro workflow"
    script "workflows/mcmicro.nf"
    workflow "MCMICRO"

    test("test 1") {

        when {
            params {
                input_sample = "$projectDir/assets/samplesheet-test.csv"
                marker_sheet = "$projectDir/assets/markers-test.csv"
            }
            workflow {
                """
                input[0] = Channel.fromList([
                    [[id:"TEST1"], ["https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/imaging/ome-tiff/cycif-tonsil-cycle1.ome.tif"]]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.versions).match("versions") }
            )
        }

    }
}
