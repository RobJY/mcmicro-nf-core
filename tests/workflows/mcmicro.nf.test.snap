{
    "versions": {
        "content": [
            [
                "versions.yml:md5,038a3138460501f8cf5dcd982b1f05aa",
                "versions.yml:md5,a95fffe435a8977d85664177b448b52c",
                "versions.yml:md5,b25dd7741700a85d5421d11318fb0e99"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-19T11:45:40.411452684"
    }
}